lib/rspec/mocks/any_instance/stub_chain.rb

```diff
@@ -30,7 +30,7 @@ def invocation_order
             :with => [nil],
             :and_return => [:with, nil],
             :and_raise => [:with, nil],
-            :and_yield => [:with, nil],
+            :and_yield => [:with, :and_yield, nil],
```

Comment:
`with` is first and so always supports further chaining, `and_yield` also supports (I think) `and_return` and `and_raise` after it, I'm sure `and_call_original` and `and_wrap_original` should have some combinations too